transport/uml/push/token_invalidated_push_flow.puml

```diff
@@ -0,0 +1,36 @@
+@startuml
+'https://plantuml.com/sequence-diagram
+
+!theme bluegray
+!pragma teoz true
+
+
+skinparam backgroundColor #whitesmoke
+title Invalidate session token
+
+box TransportSDK #lightcyan
+participant MessagingClient
+participant PushService
+participant Vault
+end box
+box Backend
+participant Shyrka
+end box
+
+		rnote left of MessagingClient
+		Since UI might have custom implementation of Vault and can remove
+		the session token at any time without Transport knowing about that.
+		This use-case is covered from within the session start flow.
+    	end rnote
+
+	Shyrka -> MessagingClient : onMessage(\n ConnectionClosedEvent.Reason.UserSignedIn  OR \n LogoutEvent OR \n ErrorEvent.ErrorCode.CannotDowngradeToUnauthenticated \n)
+    MessagingClient -> MessagingClient: invalidateSessionToken()
+    MessagingClient -> PushService : deleteDeviceToken(token))
+    MessagingClient -> Vault : remove(token)
+    MessagingClient -> Vault : getToken.
+    Vault --> Vault : TokenGenerator.generate() : token
+    Vault -> Vault : store(token)
```

Comment:
In this use case, it seems like the conversation is ended, and you remove the session token and the push notification device token. However, you then generate a new session token and store it. I wonder why? If it is needed, why not also re-register a push notification device token with the new session token?

Reply:
@KalmanPerkelGenesys We will first send request to delete deviceToken from the BE and then remove it from vault.